### PR TITLE
filtron: init at 0.2.0

### PR DESCRIPTION
**Pull request by dasj19**
###### Motivation for this change
I packaged `filtron` in order to protect a `searx` instance from bots and unwanted traffic.
I packaged its go dependencies as well.

The code I use to run filtron as a service:

```
  # Startup a filtron server.
  systemd.services.filtron-main = {
      wantedBy = [ "multi-user.target" ]; 
      after = [ "network.target" ];
      description = "Start a filtron instance.";
      serviceConfig = {
        User = "filtronm";
        ExecStart = ''
          ${pkgs.filtron}/bin/filtron -rules /etc/nixos/filtron-rules.json -target "127.0.0.1:8100"
        '';
      };
   };
```
My searx instance running as a service on port 8100 (I use a systemd service because I have two searx instances on the server)
```
  # Startup a main searx server.
  systemd.services.searx-main = {
      wantedBy = [ "multi-user.target" ]; 
      after = [ "network.target" ];
      description = "Start a main searx instance.";
      serviceConfig = {
        User = "searxm";
        ExecStart = ''
          ${pkgs.searx}/bin/searx-run
        '';
        Environment = [
          "SEARX_SETTINGS_PATH=/etc/nixos/searx-config.yml"
        ];         
      };
   };
```
Apache configuration:
```
  services.httpd.virtualHosts."searx.example".extraConfig = ''
    # Disable logs for privacy.
    SetEnvIf Request_URI "/" dontlog
    # Proxy addresses.
    ProxyPass / http://127.0.0.1:4004/
    ProxyPassReverse / http://127.0.0.1:4004/
  '';
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dasj19** [comment]
CC'ing searx's maintainers for review: @matejc @fpletz @globin @danielfullmer
Thanks.

**Comment by danielfullmer** [comment]
LGTM--It builds correctly for me, at least. I don't plan to use filtron myself, since my searx instance is private.

While a nixos module that works with the existing searx module would be nice, I don't think it should be required before merging.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please us buildGoModule.

File: pkgs/servers/filtron/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/filtron/default.nix:26]:
```suggestion
  };
}
```

File: pkgs/servers/filtron/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/filtron/default.nix:16]: No tests? Please add a comment.

Comment (dasj19) [reply pkgs/servers/filtron/default.nix:17]: I have tried to fix the tests with a substituteInPlace but it requires much more work than I anticipated, so I marked upstream tests as obsolete and therefore I turned off the tests.

**Pull request merged**
